### DISCUSSION TITLE
The loss doesn't decrease ... What can I do? (VGG16, CIFAR100 dataset)

### DISCUSSION THREAD

**Discussion by gahyunson** [community/community#55010] [Programming Help]
I try to transfer learning VGG model with CIFAR100 dataset.

I tried to resizing, randomflip, rescaling at train data,
resizing, scaling at validation, test data.

and loaded vgg16 model(include_top=False, trainable=Fasle), 
and added some custom dense layers.

The results are train loss : 4.xxx ... to 0.xxx
validation loss : 3.xxx ... to 3.xxx

Here are codes what I using.

If someone helps me, I will be really grateful.

```python
# x_data preprocess
data_transforms = {
    'TRAIN': tf.keras.Sequential([
        tf.keras.layers.experimental.preprocessing.Resizing(IMG_SIZE, IMG_SIZE),
        tf.keras.layers.experimental.preprocessing.RandomFlip("horizontal_and_vertical"),
        tf.keras.layers.experimental.preprocessing.Rescaling(scale=1./255),
        # tf.keras.layers.experimental.preprocessing.Normalization(mean=[0.5071, 0.4866, 0.4409], variance=[0.2673, 0.2564, 0.2762]),
    ]),
    'VAL': tf.keras.Sequential([
        tf.keras.layers.experimental.preprocessing.Resizing(IMG_SIZE, IMG_SIZE),
        # tf.keras.layers.experimental.preprocessing.CenterCrop(224, 224),
        tf.keras.layers.experimental.preprocessing.Rescaling(scale=1./255),
        # tf.keras.layers.experimental.preprocessing.Normalization(mean=[0.5071, 0.4866, 0.4409], variance=[0.2673, 0.2564, 0.2762]),
    ]),
}

train_dataset = tf.data.Dataset.from_tensor_slices((x_train, prey_train))
val_dataset = tf.data.Dataset.from_tensor_slices((x_val, prey_val))
test_dataset = tf.data.Dataset.from_tensor_slices((x_test, prey_test))

train_dataset = train_dataset.map(lambda x, y: (data_transforms['TRAIN'](x), y))
val_dataset = val_dataset.map(lambda x, y: (data_transforms['VAL'](x), y))
test_dataset = test_dataset.map(lambda x, y: (data_transforms['VAL'](x), y))

train_dataset = train_dataset.batch(batch_size)
val_dataset = val_dataset.batch(batch_size)
test_dataset = test_dataset.batch(batch_size)

# preprocess y data
def class_len(y):
  return len(np.unique(y.reshape(-1,)))
  
class_num = class_len(y_train)

prey_train = to_categorical(y_train, class_num)
prey_val = to_categorical(y_val, class_num)
prey_test = to_categorical(y_test, class_num)

# load vgg model
vgg_model = tf.keras.applications.vgg16.VGG16(
    include_top = False,
    weights='imagenet',
    classifier_activation='softmax'
)
for layer in vgg_model.layers:
  layer.trainable = False
```

```python
IMG_SHAPE = img_size + (3,)
com_loss = 'categorical_crossentropy'
lr = 1e-1
opt = Adam(learning_rate=lr)

model = Sequential()
model.add(vgg_model)
model.add(layers.GlobalAveragePooling2D())
model.add(layers.Flatten())
model.add(layers.Dense(128, activation='relu'))
model.add(layers.Dense(class_num, activation='softmax'))

model.compile(loss = com_loss,
              optimizer = opt,
              metrics=['acc'])

epochs = 10
val_steps = 16

tf.config.run_functions_eagerly(True)
history = model.fit(train_dataset, epochs = epochs, 
                    batch_size = 16,
                    validation_data=(val_dataset),
                    callbacks=[early_stop, Tensor_board, lr_cb],
                    validation_steps=val_steps)
```
